From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KurumiFake <kurumiisshidowife@gmail.com>
Date: Sun, 19 Jun 2022 20:05:09 +0700
Subject: [PATCH] Revert "Duplicate buf instead of Copy"

This reverts commit d33ddb90145918a36b6bcc5f949080ed71dc074e.
seems to cause some issues

diff --git a/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java b/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java
index 074d2437..3992f521 100644
--- a/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java
+++ b/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java
@@ -52,8 +52,7 @@ public class MinecraftDecoder extends MessageToMessageDecoder<ByteBuf>
             }
         }
 
-        // FlameCord - Duplicate buf instead of Copy
-        ByteBuf slice = in.duplicate(); // Can't slice this one due to EntityMap :(
+        ByteBuf slice = in.copy(); // Can't slice this one due to EntityMap :(
 
         Object packetTypeInfo = null;
         try
@@ -93,8 +92,7 @@ public class MinecraftDecoder extends MessageToMessageDecoder<ByteBuf>
                 in.skipBytes( in.readableBytes() );
             }
 
-            // FlameCord - Duplicate buf instead of Copy
-            out.add( new PacketWrapper( packet, slice.retain() ) );
+            out.add( new PacketWrapper( packet, slice ) );
             slice = null;
         } catch (BadPacketException | IndexOutOfBoundsException e) {
             // Waterfall start: Additional DoS mitigations
